subsys/bluetooth/audio/has.c

```diff
@@ -34,11 +34,96 @@ static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *at
 				 sizeof(has.features));
 }
 
+#if CONFIG_BT_HAS_PRESET_COUNT > 0
+static uint8_t handle_control_point(struct bt_conn *conn, uint8_t opcode,
+				    struct net_buf_simple *buf)
+{
+	BT_DBG("conn %p opcode %s (0x%02x)", conn, bt_has_op_str(opcode), opcode);
+
+	/* TODO: handle request here */
+
+	return BT_HAS_ERR_INVALID_OPCODE;
+}
+
+static ssize_t write_control_point(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+				   const void *data, uint16_t len, uint16_t offset, uint8_t flags)
+{
+	const struct bt_has_cp_hdr *hdr;
+	struct net_buf_simple buf;
+	uint8_t err;
+
+	BT_DBG("conn %p attr %p data %p len %d offset %d flags 0x%02x", conn, attr, data, len,
+	       offset, flags);
+
+	if (offset > 0) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+	}
+
+	if (len < sizeof(*hdr)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	net_buf_simple_init_with_data(&buf, (void *)data, len);
+
+	hdr = net_buf_simple_pull_mem(&buf, sizeof(*hdr));
+
+	err = handle_control_point(conn, hdr->opcode, &buf);
```

Comment:
Any reason why we are extracting the header here before calling `handle_control_point`, rather than doing that in `handle_control_point`?